test-runner-poc/lib/runner.js

```diff
@@ -29,18 +28,28 @@ function environmentSetup(environmentCallback, done) {
     createFilters(environmentParser);
     function environmentParser(environmentName, version) {
       const Environment = environments[environmentName];
-      client.db('admin').command({ isMaster: 1 }, (err, result) => {
+
+      function callEnvironmentCallback(result, version, err) {
+        const environment = new Environment(result, version);
+        environment.mongo = require('../../index');
+        environmentCallback(err, environment);
+        return client.close(done);
+      }
+
+      client.db('admin').command({ isMaster: 1, }, (err, result) => {
         if (err) environmentCallback(err)
```

Comment:
We should be returning here and let's use the
```
if(err) {

}
```
syntax here.

Reply:
fixed.

Reply:
Looks like this isn't fixed